XcodeServerSDK/XcodeServer.swift

```diff
@@ -186,23 +186,55 @@ public extension XcodeServer {
             completion(success: false, error: Error.withInfo("Nil response"))
         }
     }
-
-    public func createBot(botOrder: Bot, completion: (bot: Bot?, error: NSError?) -> ()) {
-        
-        let body: NSDictionary = botOrder.dictionarify()
+    
+    public enum CreateBotResponse {
+        case Success(bot: Bot)
+        case BlueprintNeedsFixing(fixedBlueprint: SourceControlBlueprint)
+        case Error(error: ErrorType)
```

Comment:
This is an enum of possible asynchronous responses. I don't know how `throws` works in asynchronous cases, since it relies on return values and current thread (AFAIK). Neither is useful in async APIs such as this one. 

But I might be missing something. Let me know if you can think of a way. 

I'll respond to the rest tomorrow, thanks a lot for taking the time to look at it. 

Sent from my iPhone

> On Jun 27, 2015, at 12:23 AM, Esteban Torres <notifications@github.com> wrote:
> 
> In XcodeServerSDK/XcodeServer.swift:
> 
> > @@ -186,23 +186,55 @@ public extension XcodeServer {
> >              completion(success: false, error: Error.withInfo("Nil response"))
> >          }
> >      }
> > -
> > -    public func createBot(botOrder: Bot, completion: (bot: Bot?, error: NSError?) -> ()) {
> > -        
> > -        let body: NSDictionary = botOrder.dictionarify()
> > +    
> > +    public enum CreateBotResponse {
> > +        case Success(bot: Bot)
> > +        case BlueprintNeedsFixing(fixedBlueprint: SourceControlBlueprint)
> > +        case Error(error: ErrorType)
> Would we want to use case Error rather than throws?
> 
> I like receiving the error case but wouldn't it brake consistency with the rest of the code (as is being envisioned for Swift 2.0?)
> 
> —
> Reply to this email directly or view it on GitHub.
>